blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/RunSearch.java

```diff
@@ -83,6 +84,40 @@ public Link getLink() {
         return runs;
     }
 
+    public static Iterable<BlueRun> findRuns(Job job, final Link parent, int start, int limit){
+        final List<BlueRun> runs = new ArrayList<>();
+        Iterable<Job> pipelines;
+        if(job != null){
+            pipelines = ImmutableList.of(job);
+        }else{
+            pipelines = Jenkins.getInstance().getItems(Job.class);
+        }
+
+        int skipCount=0;
+        int limitCount=0;
+        for (Job p : pipelines) {
+            RunList<? extends Run> runList = p.getBuilds();
```

Comment:
@kzantow Not from what I can see both looking at the code, and looking in a debugger. See [`Job.getBuilds`](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/model/Job.java#L701) ... calls `_getRuns().values()`, where `_getRuns()` returns a `SortedMap` ... that's not a lazy loading of them afaics.